its/plugin/src/integrationTest/java/org/sonarsource/iac/PropertiesTest.java

```diff
@@ -126,11 +126,28 @@ void testDeprecatedArmActivationPropertyKey(String projectKey, Boolean armProper
     assertThat(getMeasureAsInt(projectKey, "ncloc")).isEqualTo(expectedNcloc);
   }
 
+  @ParameterizedTest
+  @CsvSource(delimiter = ';', value = {
+    "springConfigCustomPattern; application*.properties,config*.properties,*.yaml; 5; 13",
+    "springConfigCustomPropertiesPattern; application*.properties,config*.properties; 4; 4",
+    "springConfigExcludeBoot; config*.properties; 2; 2",
+    "springCustomYaml; *.yaml; 1; 9",
+  })
+  void testSpringConfigPatterns(String projectKey, String patterns, int expectedFiles, int expectedNCloc) {
+    var sonarScanner = getSonarScanner(projectKey, BASE_DIRECTORY + "patterns", "java", "springconfig-its");
+    sonarScanner.setProperty("sonar.java.springconfig.file.patterns", patterns);
+
+    ORCHESTRATOR.executeBuild(sonarScanner);
+
+    assertThat(getMeasureAsInt(projectKey, "files")).isEqualTo(expectedFiles);
+    assertThat(getMeasureAsInt(projectKey, "ncloc")).isEqualTo(expectedNCloc);
```

Comment:
It looks that this can be replaced by `executeBuildAndAssertMetric()`.
It needs a few adjustments, my suggestions:
* pass the property as a whole string instead of concatenation - then a full-text search of the property will find it, what can be helpful, and will be easier to read
* `directorySuffix` -> `directory`
* new @Nullable argument for `profileName`
* `String metricKey, int expectedResultOfMetric` -> `int expectedNumberOfFiles, int expectedNCLOC` -> one execution of the sonar scanner can validate both measurements.